contracts/libraries/SwapIntention.sol

```diff
@@ -0,0 +1,68 @@
+//SPDX-License-Identifier: UNLICENSED
+
+pragma solidity ^0.8.20;
+
+import {PathKey} from "./PathKey.sol";
+import {Currency} from "@uniswap/v4-core/contracts/types/Currency.sol";
+import {IHooks} from "@uniswap/v4-core/contracts/interfaces/IHooks.sol";
+import {PoolKey} from "@uniswap/v4-core/contracts/types/PoolKey.sol";
+
+enum SwapType {
+    ExactInput,
+    ExactInputSingle,
+    ExactOutput,
+    ExactOutputSingle
+}
+
+struct SwapInfo {
+    SwapType swapType;
+    bytes params;
+}
+
+struct ExactInputSingleParams {
```

Comment:
oo also i just remembered we have these defined here https://github.com/Uniswap/swap-router-contracts-next/blob/main/contracts/interfaces/IV4SwapRouter.sol so you can just import them!

Reply:
yeah, I pulled them from the Router interface to their own file given they are not either router or quoter specific now

Reply:
My feeling is that they're still mostly _router_ specific structs, but quoter happens to also use them since quoter is trying to adhere to router interface for easy integrations. Also since router is in a separate repo, now these are duplicated and liable to diverge? I'd prefer we just import them

Reply:
fwiw router is in the same repo and will most likely be reusing the same structs so I still lean toward not putting them in IRouter? wonder what others think

Reply:
ok I think Router needs an extra field `ExactInputMaximum/ExactOutputMinimum` which makes this tricker... Maybe should just have two sets of structs for Quoter and Router respectively (this is what v3 did)